Sources/SpeziLLM/LLMRunner.swift

```diff
@@ -8,192 +8,187 @@
+/// The ``LLMRunner`` delegates the creation of the ``LLMSession``s to the respective ``LLMPlatform``s, allowing for customized creation and dependency injection for each LLM type.
 ///
-/// The runner manages a set of ``LLMGenerationTask``'s as well as the respective LLM execution backends in order to enable
-/// a smooth and efficient model execution.
+/// Within the Spezi ecosystem, the ``LLMRunner`` is set up via the Spezi `Configuration` by taking a trailing closure argument within ``LLMRunner/init(_:)``.
+/// The closure aggregates multiple stated ``LLMPlatform``s via is the ``LLMPlatformBuilder``, enabling easy and dynamic configuration of all wanted ``LLMPlatform``s.
+///
+/// The main functionality of the ``LLMRunner`` is``LLMRunner/callAsFunction(with:)``, turning a ``LLMSchema`` to an executable ``LLMSession`` via the respective ``LLMPlatform``.
+/// The created ``LLMSession`` then holds the LLM context and is able to perform the actual LLM inference.
+/// For one-shot LLM inference tasks, the ``LLMRunner`` provides ``LLMRunner/oneShot(with:chat:)``, enabling the ``LLMRunner`` to deal with the LLM state management and reducing the burden on developers by just returning an `AsyncThrowingStream`.
 ///
 /// ### Usage
 ///
-/// The code section below showcases a complete code example on how to use the ``LLMRunner`` in combination with a `LLMLocal` (locally executed Language Model) from the [SpeziLLMLocal](https://swiftpackageindex.com/stanfordspezi/spezillm/documentation/spezillm/spezillmlocal) target.
+/// The code section below showcases a complete, bare-bone code example on how to use the ``LLMRunner`` with the ``LLMSchema``.
+/// The example is structured as a SwiftUI `View` with a `Button` to trigger LLM inference via the ``LLMMockSchema``. The generated output stream is displayed in a `Text` field.
 ///
 /// ```swift
-/// class LocalLLMAppDelegate: SpeziAppDelegate {
-///     override var configuration: Configuration {
-///         Configuration {
-///             // Configure the runner responsible for executing LLMs
-///             LLMRunner(
-///                 runnerConfig: .init(
-///                     taskPriority: .medium
-///                 )
-///             ) {
-///                 // Runner setup tasks conforming to `LLMRunnerSetupTask` protocol
-///                 LLMLocalRunnerSetupTask()
+/// struct LLMDemoView: View {
+///     // The runner responsible for executing the LLM.
+///     @Environment(LLMRunner.self) var runner: LLMRunner
+///
+///     // The LLM in execution, as defined by the ``LLMSchema``.
+///     @State var llmSession: LLMMockSession?
+///     @State var responseText = ""
+///
+///     var body: some View {
+///         VStack {
+///             Button {
+///                 Task {
+///                     try await executePrompt(prompt: "Hello LLM!")
+///                 }
+///             } label: {
+///                 Text("Start LLM inference")
 ///             }
```

Comment:
Would probably make sense to disable this button if the llmSession is nil as it is still in the setup process? Might be a thing that developers should be aware of that the creation of the session might take some time and the UI should be setup accordingly?

Reply:
Good catch, I wanted to keep the example as simple as possible, but this definitely makes sense!

Reply:
I think this might even be a more general architecture question especially for longer-running setup tasks. Does it make sense to somehow add some convenience functionally in the framework that deals with that state management?

Reply:
I added the `View/disabled(llm:)` to provide a small simplification for users of the framework.